### PR TITLE
Move NLP timings to common

### PR DESCRIPTION
**Pull request by FreyJo**

### CODE REVIEW

**Review by david0oo — COMMENTED** [review]
Shouldn't we use timings instead of nlp_mem->timings? In line 663

File: acados/ocp_nlp/ocp_nlp_ddp.c
Comment (david0oo) [inline acados/ocp_nlp/ocp_nlp_ddp.c:1]: In line 663 write timings instead of nlp_mem->timings

File: acados/ocp_nlp/ocp_nlp_ddp.c [hunk 1]
```diff
@@ -1046,7 +1035,7 @@ void ocp_nlp_ddp_eval_param_sens(void *config_, void *dims_, void *opts_, void *
     ocp_nlp_common_eval_param_sens(config, dims, opts->nlp_opts, nlp_mem, nlp_work,
                                  field, stage, index, sens_nlp_out);
 
-    mem->time_solution_sensitivities = acados_toc(&timer0);
+    nlp_mem->nlp_timings->time_solution_sensitivities = acados_toc(&timer0);
```
Comment (david0oo) [inline acados/ocp_nlp/ocp_nlp_ddp.c:1018]: only `timings`?

File: acados/ocp_nlp/ocp_nlp_ddp.c [hunk 2]
```diff
@@ -671,7 +660,7 @@ int ocp_nlp_ddp(void *config_, void *dims_, void *nlp_in_, void *nlp_out_,
 
     // zero timers
     double tmp_time;
-    ocp_nlp_ddp_reset_timers(mem);
+    ocp_nlp_timings_reset(nlp_mem->nlp_timings);
```
Comment (david0oo) [inline acados/ocp_nlp/ocp_nlp_ddp.c:663]: only `timings`?

File: acados/ocp_nlp/ocp_nlp_ddp.c [hunk 3]
```diff
@@ -662,6 +650,7 @@ int ocp_nlp_ddp(void *config_, void *dims_, void *nlp_in_, void *nlp_out_,
     ocp_nlp_memory *nlp_mem = mem->nlp_mem;
     ocp_qp_xcond_solver_config *qp_solver = config->qp_solver;
     ocp_nlp_res *nlp_res = nlp_mem->nlp_res;
+    ocp_nlp_timings *timings = nlp_mem->nlp_timings;
```
Comment (david0oo) [inline acados/ocp_nlp/ocp_nlp_ddp.c:653]: Call `nlp_timings` to make consistent with SQP?

File: acados/ocp_nlp/ocp_nlp_sqp.c [hunk 4]
```diff
@@ -946,7 +931,7 @@ void ocp_nlp_sqp_eval_param_sens(void *config_, void *dims_, void *opts_, void *
     ocp_nlp_common_eval_param_sens(config, dims, opts->nlp_opts, nlp_mem, nlp_work,
                                  field, stage, index, sens_nlp_out);
 
-    mem->time_solution_sensitivities = acados_toc(&timer0);
+    nlp_mem->nlp_timings->time_solution_sensitivities = acados_toc(&timer0);
```
Comment (david0oo) [inline acados/ocp_nlp/ocp_nlp_sqp.c:921]: only `nlp_timings`

File: acados/ocp_nlp/ocp_nlp_ddp.c (hunk 3 above)
Comment (FreyJo) [reply acados/ocp_nlp/ocp_nlp_ddp.c]: done

File: acados/ocp_nlp/ocp_nlp_ddp.c (hunk 1 above)
Comment (FreyJo) [reply acados/ocp_nlp/ocp_nlp_ddp.c:1018]: Here I kept it, thought it is not worth to unpack the pointer if only one field is used.
Note that this is not in the main ddp function.

File: acados/ocp_nlp/ocp_nlp_ddp.c (hunk 2 above)
Comment (FreyJo) [reply acados/ocp_nlp/ocp_nlp_ddp.c]: doen

**Review by david0oo — APPROVED** [review]
Looks good

**Review by david0oo — APPROVED** [review]

**Pull request merged**